packages/observable/src/observable.ts

```diff
@@ -10,63 +10,171 @@ import * as dependencyDetection from './dependencyDetection'
 import { deferUpdates } from './defer'
 import { subscribable, defaultEvent, LATEST_VALUE } from './subscribable'
 import { valuesArePrimitiveAndEqual } from './extenders'
+import type { Subscribable, MaybeSubscribable } from './subscribable'
 
-export function observable (initialValue) {
-  function Observable () {
+//#region Observable
+
+/**
+ * Represents a value that can be either a plain value or an observable.
+ */
+export type MaybeObservable<T = any> = T | Observable<T>;
+
+/**
+ * Defines the functions available on an observable.
+ */
+export interface ObservableFunctions<T = any> extends Subscribable<T> {
+  /**
+   * Compares two values for equality.
+   * @param a The first value.
+   * @param b The second value.
+   * @returns True if the values are equal, otherwise false.
+   */
+  equalityComparer(a: T, b: T): boolean;
+
+  /**
+   * Returns the current value of the observable without creating a dependency.
+   * @returns The current value.
+   */
+  peek(): T;
+
+  /**
+   * Notifies subscribers that the value has changed.
+   */
+  valueHasMutated(): void;
+
+  /**
+   * Notifies subscribers that the value is about to change.
+   */
+  valueWillMutate(): void;
+
+  /**
+   * Modifies the value of the observable using a function.
+   * @param fn The function to modify the value.
+   * @param peek Whether to use the current value without creating a dependency.
+   * @returns The modified observable.
+   */
+  modify(fn, peek? : Boolean): Observable
```

Comment:
```ts
modify (fn: (v: T) => T, peek?: boolean): undefined
```

I'm _pretty sure_ it returns `undefined`

<img width="433" alt="CleanShot 2025-04-22 at 15 02 19@2x" src="https://github.com/user-attachments/assets/5f3456cf-3fb8-4d85-80b7-af20f2286dd1" />

Reply:
You're right, but then i will also remove the return statement here, right? https://github.com/knockout/tko/blob/1edce5239ab70467ea7c01c00ae7355a8c133381/packages/observable/src/observable.ts#L61

see https://github.com/Auge19/tko/pull/4